src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -248,7 +245,8 @@ public TopicManagementService(Map<String, Object> props, String serviceName) {
     _executor = Executors.newSingleThreadScheduledExecutor(new TopicManagementServiceThreadFactory());
     _configuredTopicReplicationFactor = config.getInt(TopicManagementServiceConfig.TOPIC_REPLICATION_FACTOR_CONFIG);
     _topicCreationEnabled = config.getBoolean(TopicManagementServiceConfig.TOPIC_CREATION_ENABLED_CONFIG);
-    _topicFactoryConfig = config.getString(TopicManagementServiceConfig.TOPIC_FACTORY_CONFIG);
+    String topicFactoryConfig = config.getString(TopicManagementServiceConfig.TOPIC_FACTORY_CONFIG);
+    _topicFactory = (TopicFactory) Class.forName(topicFactoryConfig).getConstructor(Map.class).newInstance(props);
```

Comment:
Having nested configuration that matches the composition of classes leaks implementation details into the configuration.  Also it's not clear what the implementations of TopicFactory may need in the future.  Limiting this to the properties seen by the topic management service is probably safe.